src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -1077,6 +1077,41 @@ struct broadcast_kernel {
   }
 };
 
+namespace
+{
+  struct shape_and_stride {
+    int32_t in_stride[MXNET_SPECIAL_MAX_NDIM];
+    int32_t out_stride[MXNET_SPECIAL_MAX_NDIM];
+    int32_t input_shape[MXNET_SPECIAL_MAX_NDIM];
+    int32_t output_shape[MXNET_SPECIAL_MAX_NDIM];
+  };
+}
+
+template<typename OP>
+struct broadcast_kernel_gpu {
+  template<typename IType, typename OType>
+  MSHADOW_XINLINE static void Map(int32_t i,
+                                  IType *input,
+                                  OType *output,
+                                  const struct shape_and_stride aux_data,
+                                  const OpReqType req,
+                                  const int32_t ndim) {
+    int32_t idx = i;
+    int32_t in_idx = i;
+#pragma unroll 4
```

Comment:
Did you benchmark if unroll makes a measurable difference? Background: currently libmxnet.so is quite large, and we may want to remove loop unrolling and force inline to reduce the size

Reply:
the loop has only 4 iterations. Plus at @ptrendx suggestion that when using array from struct compiler doesn't know the index at compile time.
So unroll can give better performance. 

@leezu To answer your question : yes, there is improvement in performance when unrolling slightly more when `unroll 4`compared to `unroll`

Reply:
@access2rohit Could you provide data. How much performance improvement with unroll and unroll 4?

Also the index_t in this kernel is still int64_t?

Reply:
@apeforest 
The performance gain is about 10-12% with unroll. 
I wanted to keep the implementation consistent with existing code so I kept it index_t. Since the target was to make sure LTS perf >= master no-LTS performance with the new approach it still there.

I can also hard code every variable inside GPU kernel to be int32_t and there will be 10-15% further gain for LTS MXNet.

Reply:
It is confusing to mix index_t with int in this kernel since your variable `i` in the Map function is `int`

Reply:
aah  ... my bad will fix it